src/workflow/cache_manager.rs

```diff
@@ -0,0 +1,154 @@
+use crate::workflow::WorkflowCachingPolicy;
+use lru::LruCache;
+use std::collections::HashSet;
+
+pub(crate) struct WorkflowCacheManager {
+    /// LRU cache of run IDs.
+    cache: LruCache<String, bool>,
```

Comment:
Value should be `()` rather than a bool, slightly smaller and avoids possible `false` values.

Reply:
Done